lib/internal/quic/datagrams.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+const {
+  Uint8Array,
+} = primordials;
+
+const {
+  ReadableStream,
+} = require('internal/webstreams/readablestream');
+
+const {
+  WritableStream,
+} = require('internal/webstreams/writablestream');
+
+const {
+  isArrayBufferView,
+} = require('util/types');
+
+const {
+  codes: {
+    ERR_INVALID_ARG_TYPE,
+  },
+} = require('internal/errors');
```

Comment:
@mcollina ... a question for you in this. The intent of the code in this file is to start providing an interface similar to the standard [`WebTransportDatagramDuplexStream`](https://developer.mozilla.org/en-US/docs/Web/API/WebTransportDatagramDuplexStream) API ... however, this starts getting into the territory of the higher level APIs (in case we wanted to implement `WebTransport` separately in a dependency like undici instead, for instance). The question is whether our internal low level APIs should be based on web streams here or should we expose only the most minimal low-level primitives at this level.

Reply:
Actually... the more I think about it, the more I want to pull this back to a more minimal interface and let the APIs that are building on top (like `WebTransport`) handle the interface to web streams. So I'm going to pull this back into an event-based API instead and eliminate the datagram.js file here.